### DISCUSSION TITLE
Issue with alert labels in n9e

### DISCUSSION THREAD

**Discussion by hdsfgsgrlsgtw12** [ccfos/nightingale#1286] [Q&A]
我查询promsql的时候查询的标签只有两个instance和queue_name
![image](https://user-images.githubusercontent.com/45323722/203705808-51be2ac8-62f0-418c-bc6a-773ce452bccb.png)
但是告警告出来的标签还包括了_name_和rulename如下，这两个是内置标签吧？我想要告警看起来简单一点，请教一下两个内置标签能否去掉，如果可以，怎么去掉呢？
![image](https://user-images.githubusercontent.com/45323722/203706087-ad9328ab-52c5-42d7-ac59-47c70b3869af.png)

**Comment by wushengwusheng** [comment]
n9e服务端应该有个./etc/script/notify.py脚本 改下应该就可以吧
<img width="457" alt="image" src="https://user-images.githubusercontent.com/30250875/203749288-3e67addd-7a3d-42c0-a6d5-a1b30c6cf56b.png">

**Reply by hdsfgsgrlsgtw12** [reply]
没看见有这个脚本呢
![image](https://user-images.githubusercontent.com/45323722/203908528-a1058d53-1b8f-4816-8514-8e5273850301.png)

**Comment by UlricQin** [comment]
`__name__`就是metric名字，在prometheus内部，metric也是一个特殊的标签，标签key是`__name__`，如果不想展示，可以修改etc/templates下面的几个tpl文件，那些tpl文件控制着推送的告警消息长什么样子，可以对TagsJSON循环遍历，不想展示的字段就continue一下，这些tpl都遵从go template语法

**Reply by hdsfgsgrlsgtw12** [reply]
遍历是在下面图片tpl模板中，还是在告警通知notify.py里面遍历啊，notify.py这个我都没找到这个脚本
![image](https://user-images.githubusercontent.com/45323722/203917199-f74da773-f7b2-4c6b-9f7c-f971168fecc5.png)
![image](https://user-images.githubusercontent.com/45323722/203917412-120f74de-63ff-4bb5-829d-3dcd98a6f57e.png)

**Reply by FengZh61** [reply]
> 遍历是在下面图片tpl模板中，还是在告警通知notify.py里面遍历啊，notify.py这个我都没找到这个脚本 ![image](https://user-images.githubusercontent.com/45323722/203917199-f74da773-f7b2-4c6b-9f7c-f971168fecc5.png) ![image](https://user-images.githubusercontent.com/45323722/203917412-120f74de-63ff-4bb5-829d-3dcd98a6f57e.png)

notify.py是拓展用的，本身就支持的通知方式用不到的，而且这个是在n9e的路径下，n9e/etc/script/*

**Reply by hdsfgsgrlsgtw12** [reply]
这不是一个目录，vim进去是乱码
![image](https://user-images.githubusercontent.com/45323722/203948881-f3910d86-3837-4170-83ac-18779abb8e77.png)
对TagsJSON循环遍历的话，是在这个脚本下进行嘛？还是我自己写代码进行遍历呢？

**Reply by FengZh61** [reply]
> 这不是一个目录，vim进去是乱码 ![image](https://user-images.githubusercontent.com/45323722/203948881-f3910d86-3837-4170-83ac-18779abb8e77.png) 对TagsJSON循环遍历的话，是在这个脚本下进行嘛？还是我自己写代码进行遍历呢？

TagsJSON循环遍历是写在代码里的，不是脚本完成的。

**Reply by hdsfgsgrlsgtw12** [reply]
> > 这不是一个目录，vim进去是乱码 ![image](https://user-images.githubusercontent.com/45323722/203948881-f3910d86-3837-4170-83ac-18779abb8e77.png) 对TagsJSON循环遍历的话，是在这个脚本下进行嘛？还是我自己写代码进行遍历呢？
> 
> TagsJSON循环遍历是写在代码里的，不是脚本完成的。

方便加下微信具体说一下吗

**Comment by jayluo667** [comment]
> 我查询promsql的时候查询的标签只有两个instance和queue_name ![image](https://user-images.githubusercontent.com/45323722/203705808-51be2ac8-62f0-418c-bc6a-773ce452bccb.png) 但是告警告出来的标签还包括了_name_和rulename如下，这两个是内置标签吧？我想要告警看起来简单一点，请教一下两个内置标签能否去掉，如果可以，怎么去掉呢？ ![image](https://user-images.githubusercontent.com/45323722/203706087-ad9328ab-52c5-42d7-ac59-47c70b3869af.png)

楼主应该多查看官方文档：http://n9e.flashcat.cloud/docs/usage/format/，根据自己使用的是啥媒介对应改template/xxx.tpl
现在新版本，还支持告警规则名称和备注里添加变量
